routers/web/repo/pull.go

```diff
@@ -887,8 +887,6 @@ func viewPullFiles(ctx *context.Context, specifiedStartCommit, specifiedEndCommi
 		}
 
 		if pull.HeadRepo != nil {
-			ctx.Data["SourcePath"] = pull.HeadRepo.Link() + "/src/commit/" + endCommitID
```

Comment:
I think we don't need to edit `SourcePath` here. Its value is already set by the `setCompareContext`.

https://github.com/go-gitea/gitea/blob/8bee7fcf7e214ace5e4835556bfb0f96ae3d20fb/routers/web/repo/pull.go#L834

And if we change it, the maintainer of the base repository won't be able to view the file.